sklearn/preprocessing/_data.py

```diff
@@ -106,6 +106,8 @@ def _handle_zeros_in_scale(scale, copy=True, constant_mask=None):
         return scale
     # scale is an array
     else:
+        # TODO instead of checking for array, check for array API compliance, how?
+        # elif isinstance(scale, np.ndarray):
```

Comment:
I think I commented the line

> elif isinstance(scale, np.ndarray):

which would fail for any other array type. Instead of this check, we should have a check for array API compatibility of `scale`.

Reply:
We don't need to do any kind of type-based input validation in such private helper functions. Input validation is typically performed at the beginning of public functions or methods (e.g. by calling `_validate_data` in the `fit`, `transform` or `predict` methods of estimators).